packages/daemon/src/mail.js

```diff
@@ -180,69 +118,43 @@ export const makeMailboxMaker = ({
       ) {
         throw new Error(`Invalid request number ${q(messageNumber)}`);
       }
-      const req = messages.get(messageNumber);
-      const resolveRequest = resolvers.get(req);
-      if (resolveRequest === undefined) {
-        throw new Error(`No pending request for number ${messageNumber}`);
+      const message = messages.get(messageNumber);
+      if (message === undefined) {
+        throw new Error(`Invalid request, ${q(messageNumber)}`);
       }
       const id = petStore.identifyLocal(resolutionName);
       if (id === undefined) {
         throw new TypeError(
           `No formula exists for the pet name ${q(resolutionName)}`,
         );
       }
-      resolveRequest(id);
+      // TODO validate shape of request
+      const req = /** @type {import('./types.js').Request} */ (message);
+      const { responder } = E.get(req);
+      E.sendOnly(responder).respondId(id);
     };
 
     // TODO test reject
     /** @type {import('./types.js').Mail['reject']} */
-    const reject = async (messageNumber, message = 'Declined') => {
-      const req = messages.get(messageNumber);
-      if (req !== undefined) {
-        const resolveRequest = resolvers.get(req);
-        if (resolveRequest === undefined) {
-          throw new Error(`panic: a resolver must exist for every request`);
-        }
-        resolveRequest(harden(Promise.reject(harden(new Error(message)))));
+    const reject = async (messageNumber, reason = 'Declined') => {
+      const message = messages.get(messageNumber);
+      if (message !== undefined) {
+        // TODO verify that the message is a request.
+        const req = /** @type {import('./types.js').Request} */ (message);
+        const { responder } = E.get(req);
+        E.sendOnly(responder).respondId(
+          harden(Promise.reject(harden(new Error(reason)))),
+        );
       }
     };
 
-    /** @type {import('./types.js').Mail['receive']} */
-    const receive = (
-      senderId,
-      strings,
-      edgeNames,
-      ids,
-      receiverId = selfId,
-    ) => {
-      deliver({
-        type: /** @type {const} */ ('package'),
-        strings,
-        names: edgeNames,
-        formulas: ids,
-        who: senderId,
-        dest: receiverId,
-      });
-    };
```

Comment:
Subsumed `receive` into `deliver`. It wasn’t doing much and `deliver` serves for `send` as well.